reactome_analysis_api/reactome_analysis_api/controllers/datasets_controller.py

```diff
@@ -199,3 +195,9 @@ def load_data(resourceId, parameters):  # noqa: E501
     except (socket.timeout, socket.gaierror) as e:
         LOGGER.error("Socket timeout connecting to storage or queuing system: " + str(e))
         abort(503, "Failed to connect to downstream system. Please try again in a few minutes.")
+
+
+def get_result_by_keyword(keyword: str, species: str):
```

Comment:
Note: We need to adapt the API schema and add an end-point in the API for this function.